### PR TITLE
Pass bytes to yara

### LINKED ISSUE #15574: Pass bytes to yara instead of string
We are decoding the bytes in add-on files before sending them to yara. That can lead to problems. Let's not decode them and send the bytes to yara directly.

### PR DESCRIPTION
**Pull request by wagnerand**
Fix #15574.

### CODE REVIEW

File: src/olympia/scanners/tests/test_tasks.py [hunk 1]
```diff
@@ -612,6 +613,72 @@ def test_does_not_run_when_results_contain_errors(self, yara_compile_mock):
...
+        assert len(yara_results) == 1
+        yara_result = yara_results[0]
+        assert yara_result.upload == self.upload
+        assert len(yara_result.results) == 1
+        assert yara_result.results[0] == {
+            'rule': rule.name,
+            'tags': [],
+            'meta': {'filename': 'img.png'},
+        }
+        assert incr_mock.called
```
Comment (willdurand) [inline src/olympia/scanners/tests/test_tasks.py:643]: We don't necessarily need to mock/assert statsd stuff, let's keep this test case focused on one thing: reading binary content correctly when the switch is enabled.

File: src/olympia/scanners/tests/test_tasks.py [hunk 2]
```diff
@@ -612,6 +613,72 @@ def test_does_not_run_when_results_contain_errors(self, yara_compile_mock):
...
+
+        received_results = run_yara(self.results, self.upload.pk)
+
+        yara_result = ScannerResult.objects.all()[0]
+        # We don't match here, as decoding the bytes leads to missing the rule.
+        assert yara_result.results == []
+
+        # The task should always return the results.
+        assert received_results == self.results
+        assert incr_mock.called
```
Comment (willdurand) [inline src/olympia/scanners/tests/test_tasks.py:678]: Same here.

File: src/olympia/scanners/tests/test_tasks.py [hunk 3]
```diff
@@ -612,6 +613,72 @@ def test_does_not_run_when_results_contain_errors(self, yara_compile_mock):
         # The task should always return the results.
         assert received_results == self.results
 
+    @mock.patch('olympia.scanners.tasks.statsd.incr')
+    @override_switch('yara-read-binary', active=True)
+    def test_run_in_binary_mode(self, incr_mock):
+        self.upload = self.get_upload('webextension_with_image.zip')
+
+        assert len(ScannerResult.objects.all()) == 0
+        # This rule will match for all files in the xpi.
```
Comment (willdurand) [inline src/olympia/scanners/tests/test_tasks.py:622]: That's not correct, I believe.

File: src/olympia/scanners/tests/test_tasks.py [hunk 4]
```diff
@@ -612,6 +613,72 @@ def test_does_not_run_when_results_contain_errors(self, yara_compile_mock):
...
+        # The task should always return the results.
+        assert received_results == self.results
+
+    @mock.patch('olympia.scanners.tasks.statsd.incr')
+    @override_switch('yara-read-binary', active=False)
+    def test_run_in_non_binary_mode_with_binary(self, incr_mock):
+        self.upload = self.get_upload('webextension_with_image.zip')
+
+        assert len(ScannerResult.objects.all()) == 0
+        # This rule will match for all files in the xpi.
```
Comment (willdurand) [inline src/olympia/scanners/tests/test_tasks.py:661]: That's not correct, I believe.

**Review by willdurand — APPROVED** [review]
r+wc

File: src/olympia/scanners/tests/test_tasks.py (hunk 3 above)
Comment (wagnerand) [reply src/olympia/scanners/tests/test_tasks.py:622]: Oh, right, I had copied and pasted the entire test.

File: src/olympia/scanners/tests/test_tasks.py (hunk 4 above)
Comment (wagnerand) [reply src/olympia/scanners/tests/test_tasks.py:661]: Same here: I had copied and pasted the entire test.

File: src/olympia/scanners/tests/test_tasks.py (hunk 1 above)
Comment (wagnerand) [reply src/olympia/scanners/tests/test_tasks.py]: e2bfd9c225

File: src/olympia/scanners/tests/test_tasks.py (hunk 2 above)
Comment (wagnerand) [reply src/olympia/scanners/tests/test_tasks.py]: e2bfd9c225

File: src/olympia/scanners/tests/test_tasks.py (hunk 3 above)
Comment (wagnerand) [reply src/olympia/scanners/tests/test_tasks.py]: e2bfd9c225

File: src/olympia/scanners/tests/test_tasks.py (hunk 4 above)
Comment (wagnerand) [reply src/olympia/scanners/tests/test_tasks.py]: e2bfd9c225

**Pull request merged**